docs/tutorials/custom_segmentation_trainer.ipynb

```diff
@@ -0,0 +1,524 @@
+    "                    task='multiclass',\n",
+    "                    num_classes=num_classes,\n",
+    "                    beta=1.0,\n",
+    "                    average='micro',\n",
+    "                ),\n",
+    "                'MeanIoU': JaccardIndex(\n",
+    "                    num_classes=num_classes, task='multiclass', average='macro'\n",
+    "                ),\n",
+    "            },\n",
+    "            prefix='train_',\n",
+    "        )\n",
+    "        self.val_metrics = self.train_metrics.clone(prefix='val_')\n",
+    "        self.test_metrics = self.train_metrics.clone(prefix='test_')\n",
+    "\n",
+    "    def configure_callbacks(self) -> Sequence[Callback] | Callback:\n",
+    "        \"\"\"Initialize callbacks for saving the best and latest models.\n",
+    "\n",
+    "        Returns:\n",
+    "            List of callbacks to apply.\n",
+    "        \"\"\"\n",
+    "        return [\n",
+    "            ModelCheckpoint(every_n_epochs=50, save_top_k=-1, save_last=True),\n",
+    "            ModelCheckpoint(monitor=self.monitor, mode=self.mode, save_top_k=5),\n",
+    "        ]\n",
+    "\n",
+    "    def on_train_epoch_start(self) -> None:\n",
+    "        \"\"\"Log the learning rate at the start of each training epoch.\"\"\"\n",
+    "        optimizers = self.optimizers()\n",
+    "        if isinstance(optimizers, list):\n",
+    "            lr = optimizers[0].param_groups[0]['lr']\n",
+    "        else:\n",
+    "            lr = optimizers.param_groups[0]['lr']\n",
+    "        self.logger.experiment.add_scalar('lr', lr, self.current_epoch)  # type: ignore"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Train model\n",
+    "\n",
+    "The remainder of the turial is straightforward and follows the typical [PyTorch Lightning](https://lightning.ai/) training routine. We instantiate a `DataModule` for the LandCover.AI 100 dataset (a small version of the LandCover.AI dataset for notebook testing), instantiate a `CustomSemanticSegmentationTask` with a U-Net and ResNet-50 backbone, then train the model using a Lightning trainer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dm = LandCoverAI100DataModule(root='data', batch_size=10, num_workers=2, download=True)\n",
+    "\n",
+    "# You can use the following for actual training runs\n",
+    "# from torchgeo.datamodules import LandCoverAIDataModule\n",
+    "# dm = LandCoverAIDataModule(root='data', batch_size=64, num_workers=8, download=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Downloading: \"https://download.pytorch.org/models/resnet50-19c8e357.pth\" to /home/davrob/.cache/torch/hub/checkpoints/resnet50-19c8e357.pth\n",
```

Comment:
I usually strip the outputs to keep the diffs small. But I know you disagree with this approach.

Reply:
Stripped